File: cpp_questions/binary_trees/bfs.cpp
#include <iostream>
#include <vector>
#include <set>
#include <queue>

using namespace std;

struct graphNode {
  int val;

  set<graphNode* > neighbors;
  graphNode(int x) : val(x) {}
};


void bfs(graphNode* root)
{
  queue<graphNode* > graphQueue;
  graphQueue.push(root);
  while (!graphQueue.empty())
  {
    graphNode* currentNode = graphQueue.front();

    cout << "processing Node: " << currentNode->val << endl;

    for (auto node : currentNode->neighbors)
    {
      graphQueue.push(node);
    }

    graphQueue.pop();

  }
}



int main() {
  
  graphNode* test = new graphNode(1);

  for (int i = 1; i < 5; i++)
  {
    test->neighbors.insert(new graphNode(i));
  }

  for (auto node : test->neighbors)
  {
    for (int i = 1; i < 5; i++)
    {
      node->neighbors.insert(new graphNode(i));
    }
  }

  bfs(test);

  return 0;

}

File: python/python_questions/bfs.py
from queue import Queue

class graphNode:

  def __init__(self, val):
    self.val = val
    self.neighbors = set()
  
  def addNode(self, new_val):
    self.neighbors.add(graphNode(new_val))

  def bfs(self):
    graphQueue = Queue()
    graphQueue.put(self)

    while not graphQueue.empty():
      current = graphQueue.get()
      
      print(current.val)

      for node in current.neighbors:
        graphQueue.put(node)
      

root = graphNode(1)

for i in range(2,6):
  root.addNode(i)


for node in root.neighbors:
  for i in range(54,60):
    node.addNode(i)

root.bfs()